src/components/Dao/DaoWiki.tsx

```diff
@@ -0,0 +1,167 @@
+import * as React from "react";
+import * as Sticky from "react-stickynode";
+import { IDAOState, ISchemeState, Scheme, IProposalType, Proposal } from "@daostack/client";
+import { WikiContainer, actualHash, ReactiveWiki } from "@dorgtech/daosmind";
+import classNames from "classnames";
+import { enableWalletProvider } from "arc";
+import { combineLatest } from "rxjs";
+
+import { Link } from "react-router-dom";
+import * as arcActions from "actions/arcActions";
+import { showNotification } from "reducers/notifications";
+import { schemeName } from "lib/util";
+import { connect } from "react-redux";
+import { RouteComponentProps } from "react-router-dom";
+import withSubscription, { ISubscriptionProps } from "components/Shared/withSubscription";
+import Loading from "components/Shared/Loading";
+import * as proposalStyle from "../Scheme/SchemeProposals.scss";
+import * as daoStyle from "./Dao.scss";
+
+type IExternalProps = {
+  daoState: IDAOState;
+  match: Record<string, any>;
+} & RouteComponentProps<any>;
+
+const mapDispatchToProps = {
+  createProposal: arcActions.createProposal,
+  showNotification,
+};
+
+interface IDispatchProps {
+  createProposal: typeof arcActions.createProposal;
+  showNotification: typeof showNotification;
+}
+
+type SubscriptionData = ISubscriptionProps<[Scheme[], Proposal[]]>
+type IProps = IDispatchProps & IExternalProps & SubscriptionData;
+
+function DaoWiki(props: IProps) {
+  const [hasWikiScheme, setHasWikiScheme] = React.useState<boolean>(false);
+  const [schemes, proposals] = props.data;
+
+  const renderWikiComponent = () => {
+    const { daoAvatarAddress, perspectiveId, pageId } = props.match.params;
+    actualHash["dao"] = daoAvatarAddress;
+    actualHash["wiki"] = perspectiveId;
+    actualHash["page"] = pageId;
+    return WikiContainer.getInstance({});
+  };
+
+  const checkIfWikiSchemeExists = async () => {
+    const genericSchemes = schemes.filter((scheme: Scheme) => scheme.staticState.name === "GenericScheme")
+    const states: ISchemeState[] = [];
+    const getSchemeState = () => {
```

Comment:
don't you already have the scheme states from when you fetch dao.schemes and set fetchAllData to true?

Reply:
Yes! That's what we thought, but it only retrieves the "Generic" name of the scheme, and we need the real name of it (In this case `WikiUpdate`), so we found a solution by getting every state and call the function `schemeName`. 
Here's an example of the variable `schemes` that we're receiving
<img width="1042" alt="Screenshot 2020-01-18 at 00 06 49" src="https://user-images.githubusercontent.com/21031138/72652294-77691e00-3986-11ea-9c98-879b1e052678.png">

If you think this can/should be improved just let us know